src/DataPolicyConsentManager.php

```diff
@@ -69,148 +69,130 @@ public function needConsent() {
 
-    $state = $this->getState();
-    $name = $this->entity->getName() ?: $this->t('Data policy');
-
-    $link = Link::createFromRoute(strtolower($name), 'data_policy.data_policy', [], [
-      'attributes' => [
-        'class' => ['use-ajax'],
-        'data-dialog-type' => 'modal',
-        'data-dialog-options' => Json::encode([
-          'title' => $name,
-          'width' => 700,
-          'height' => 700,
-        ]),
-      ],
-    ]);
+    $entity_ids = $this->getEntityIdsFromConsentText();
+    $revisions = $this->getRevisionsByEntityIds($entity_ids);
+    $links = [];
+
+    foreach ($revisions as $key => $revision) {
+      $name = $revision->getName() ?: $this->t('Data policy');
+      $links[$key] = Link::createFromRoute(strtolower($name), 'data_policy.data_policy', ['id' => $revision->id()], [
+        'attributes' => [
+          'class' => ['use-ajax'],
+          'data-dialog-type' => 'modal',
+          'data-dialog-options' => Json::encode([
+            'title' => $name,
+            'width' => 700,
+            'height' => 700,
+          ]),
+        ],
+      ]);
+    }
 
     $enforce_consent = !empty($this->getConfig('enforce_consent'));
+    $enforce_consent_text = $this->getConfig('consent_text');
+
+    foreach ($links as $entity_id => $link) {
+      $enforce_consent_text = str_replace("[id:{$entity_id}]", $link->toString(), $enforce_consent_text);
+    }
 
     $form['data_policy'] = [
       '#type' => 'checkbox',
-      '#title' => $this->t('I agree with the @url', [
-        '@url' => $link->toString(),
-      ]),
-      '#default_value' => $state == UserConsentInterface::STATE_AGREE,
+      '#title' => $enforce_consent_text,
       '#required' => $enforce_consent && $this->currentUser->isAnonymous(),
     ];
   }
 
   /**
    * {@inheritdoc}
    */
-  public function saveConsent($user_id, $state = UserConsentInterface::STATE_UNDECIDED) {
+  public function saveConsent($user_id, $state = UserConsentInterface::STATE_UNDECIDED, $action = NULL) {
     if ($state === TRUE) {
       $state = UserConsentInterface::STATE_AGREE;
     }
     elseif ($state === FALSE) {
       $state = UserConsentInterface::STATE_NOT_AGREE;
     }
 
-    $last_state = $this->getState();
-
-    if ($last_state !== FALSE) {
-      if ($last_state == $state) {
-        return;
-      }
-      else {
-        if (!empty($this->getConfig('enforce_consent'))) {
-          // Allow switching to state with higher priority (from "undecided" to
-          // "no agree").
-          if ($last_state > $state) {
-            return;
-          }
-        }
-        else {
-          // Allow all switching cases without cases when switchbacking to the
-          // "undecided" state (from "not agree" to "undecided" or from "agree"
-          // to "undecided").
-          if ($last_state != UserConsentInterface::STATE_UNDECIDED && $state == UserConsentInterface::STATE_UNDECIDED) {
-            return;
-          }
-        }
-      }
-    }
```

Comment:
Yes, later I can explain why.

Reply:
So, this `getState` works only when you use one entity.

The method is used to get the last state. There are 3 states in total (when the user clicks on the form / data-policy-agreement = 0, when he refuses to agree = 1, and when he agrees = 3). With the current implementation, such a method does not make sense, since these data are not used anywhere. And the previous implementation created a lot of extra entries in the database.

With the current implementation, the values ​​2 and 1 are written to the database, but now we do not create an entry in the database if the user just visited the / data-policy-agreement page